src/client.c

```diff
@@ -272,16 +274,22 @@ bool set_pool(PgSocket *client, const char *dbname, const char *username, const
 		return false;
 	}
 
+	if (cf_auth_type == AUTH_HBA)
```

Comment:
This information is already available in `client->client_auth_type`. No need to look it up again.